plugins/services/src/js/containers/tasks/TasksContainer.js

```diff
@@ -77,6 +102,70 @@ class TasksContainer extends React.Component {
     this.setState({ modal: {} });
   }
 
+  handleExpressionChange(filterExpression) {
+    const { router } = this.context;
+    const { location: { pathname } } = this.props;
+    router.push({ pathname, query: { q: filterExpression.value } });
+
+    this.setState({ filterExpression });
+  }
+
+  propsToState(props) {
+    const {
+      defaultFilterData: { regions },
+      defaultFilterData: { zones }
+    } = this.state;
+
+    let query = props.location.query["q"];
+
+    if (query === undefined) {
+      query = "is:active";
+    }
+
+    const newZones = Array.from(
+      new Set(
+        props.tasks.map(task => {
+          const node = TaskUtil.getNode(task);
+
+          return node.getZoneName();
+        })
+      )
+    );
+
+    const newRegions = Array.from(
+      new Set(
+        props.tasks.map(task => {
+          const node = TaskUtil.getNode(task);
+
+          return node.getRegionName();
+        })
+      )
+    );
+
+    // If no region/ zones added from props return
+    if (
+      newRegions.length === regions.length &&
+      newRegions.every(region => regions.indexOf(region) !== -1) &&
+      newZones.length === zones.length &&
+      newZones.every(zone => zones.indexOf(zone) !== -1)
+    ) {
+      return;
+    }
+
+    const filters = new DSLFilterList([
+      new TasksStatusFilter(),
+      new TasksZoneFilter(newZones),
+      new TasksRegionFilter(newRegions),
+      new TaskNameTextFilter()
+    ]);
+
+    this.setState({
+      filterExpression: new DSLExpression(query),
+      filters,
+      defaultFilterData: { regions: newRegions, zones: newZones }
+    });
+  }
+
   fetchData() {
```

Comment:
Can we delete this line please, whenever we see something small and not in use like this we should be able to remove without opening a new PR or creating new Jira for this.